### PR TITLE
Cleanup and move to new API in C

### LINKED ISSUE #91: Cleanup old code
Remove C# code completely and create a new clean project in C/C++.

### PR DESCRIPTION
**Pull request by mcmikecreations**
The new repository is set up as a collection of subprojects in different languages.

`libs/c` contains the portable C libraries that will have bindings in other languages. For example, `cl_core` has C# bindings in `libs/csharp/common`.

`libs/cpp` contains the C++ implementations, including a common language interface `cl_lang` and the TS implementation `cl_lang_typescript`.

`libs/csharp` contains the C# bindings for the C libraries.

`libs/refs` contains reference implementations in native languages. For example, `cl_lang_csharp_ref` implements the C# language plugin using Roslyn.

`tests` contains different tests for the repository, linked to the CMake CTest tool.

`examples` contains examples in different languages, trying out various features of the repository.

`docs` contains documentation for the new repository. Old documentation has been moved to `docs/old`.

Fixes #91.

### CODE REVIEW

File: libs/c/core/include/core/language.h [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+/**
+ * @brief Transform a Crosslight node tree to a different form.
+ * 
+ * @param[in] context The initialized language context.
+ * @param[in] node Node tree to transform.
+ * @return A transformed Crosslight node tree or `0` on error.
+ * 
+ * @warning The caller is responsible for deleting the produced @ref cl_node, @p context and @p node.
+ */
+CL_API const struct cl_node* language_transform_modify(const void* context, const struct cl_node* node);
```
Comment (SergiyIvan) [inline libs/c/core/include/core/language.h:72]: What is the purpose of this method? How is it supposed to be used?

File: libs/c/core/include/core/node.h [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+CL_BEGIN_C_DECLS
+
+/**
+ * @brief Base type for language node trees.
+ */
+struct cl_node {
+    const void* payload; /**< Payload node containing specialized fields. */
+    size_t payload_type; /**< Type of the payload contained in this node. Should use values from @ref cl_node_type. */
+    const struct cl_node* parent; /**< Parent of the current node. Used for faster traversing. */
+    const struct cl_node* children; /**< Children of the current node. */
```
Comment (SergiyIvan) [inline libs/c/core/include/core/node.h:25]: Perhaps I am missing something, but shouldn't this be a _collection_ of child nodes?

**Review by SergiyIvan — APPROVED** [review]
I left a couple of comments, overall looks good, thanks!

File: libs/c/core/include/core/language.h (hunk 1 above)
Comment (mcmikecreations) [reply libs/c/core/include/core/language.h:72]: As the documentation suggests, it is meant to transform one node tree into another, e.g. for optimizations.

File: libs/c/core/include/core/node.h (hunk 2 above)
Comment (mcmikecreations) [reply libs/c/core/include/core/node.h:25]: The collection is represented as a pointer

**Comment by mcmikecreations** [comment]
@SergiyIvan please review the responses

**Comment by SergiyIvan** [comment]
> @SergiyIvan please review the responses

Thank you for your answers!

**Pull request closed without merging**